code/datums/status_effects/debuffs/debuffs.dm

```diff
@@ -1322,3 +1322,289 @@
 	REMOVE_TRAIT(owner, TRAIT_DISCOORDINATED_TOOL_USER, "[type]")
 	return ..()
+
+/datum/status_effect/fire_handler
+	duration = -1
+	alert_type = null
+	status_type = STATUS_EFFECT_MULTIPLE //Custom code
+	on_remove_on_mob_delete = TRUE
+	/// Current amount of stacks we have
+	var/stacks
+	/// Maximum of stacks that we could possibly get
+	var/stack_limit = 20
+	/// What status effect type do we merge with
+	var/our_type
+	/// What status effect type do we remove uppon being applied
+	var/enemy_type
+
+/datum/status_effect/fire_handler/on_creation(mob/living/new_owner, new_stacks, override_type = NO_FIRE_OVERRIDE, forced = FALSE)
+	owner = new_owner
+	stacks = new_stacks
+
+	var/datum/status_effect/fire_handler/our_status = owner.has_status_effect(our_type)
+	var/datum/status_effect/fire_handler/enemy_status = owner.has_status_effect(enemy_type)
+
+	if(enemy_status)
+		stacks -= enemy_status.stacks
+		enemy_status.stacks -= new_stacks
+		if(enemy_status.stacks <= 0)
+			owner.remove_status_effect(enemy_type)
+
+		if(stacks <= 0)
+			qdel(src)
+			return
+
+	if(our_status)
+		switch(override_type)
+			if(NO_FIRE_OVERRIDE)
+				if(forced)
+					our_status.stacks = min(stacks, stack_limit)
+				else
+					our_status.stacks = min(our_status.stacks + stacks, stack_limit)
+				qdel(src)
+				return
+			if(CONDITIONAL_FIRE_OVERRIDE)
+				if(stacks > our_status.stacks)
+					if(forced)
+						stacks = min(stacks, stack_limit)
+					else
+						stacks = min(stacks + our_status.stacks, stack_limit)
+					on_reapplying_stacks(our_status)
+					owner.remove_status_effect(our_type)
+				else
+					if(forced)
+						our_status.stacks = min(stacks, stack_limit)
+					else
+						our_status.stacks = min(our_status.stacks + stacks, stack_limit)
+					qdel(src)
+					return
+			if(FORCE_FIRE_OVERRIDE)
+				if(forced)
+					stacks = min(stacks, stack_limit)
+				else
+					stacks = min(stacks + our_status.stacks, stack_limit)
+				on_reapplying_stacks(our_status)
+				owner.remove_status_effect(our_type)
+
+	return ..()
+
+/datum/status_effect/fire_handler/proc/on_reapplying_stacks(datum/status_effect/fire_handler/our_status)
+
+/datum/status_effect/fire_handler/proc/update_overlay()
+
+/datum/status_effect/fire_handler/fire_stacks
+	id = "fire_stacks" //fire_stacks and wet_stacks should have different IDs or else has_status_effect won't work
+
+	our_type = /datum/status_effect/fire_handler/fire_stacks
+	enemy_type = /datum/status_effect/fire_handler/wet_stacks
+
+	/// If we're on fire
+	var/on_fire = FALSE
+	/// Type for our alert
+	var/fire_alert_type = /atom/movable/screen/alert/fire
+	/// A weakref to the mob light emitter
+	var/datum/weakref/firelight_ref
+	/// Type of mob light emitter we use when on fire
+	var/firelight_type = /obj/effect/dummy/lighting_obj/moblight/fire
+	/// Stores current fire overlay icon state, for optimisation purposes
+	var/last_icon_state
+
+/datum/status_effect/fire_handler/fire_stacks/tick(delta_time, times_fired)
```

Comment:
You're changing fire from ticking every 2 seconds (`Life()` ticks roughly every two seconds) to ticking every 1 second. You need to either set the `tick_interval` to `2 SECONDS` or half all the values associated.